src/double/main.c

```diff
@@ -0,0 +1,46 @@
+#include <stdint.h>
+#include <stdio.h>
+
+typedef union {
+    double d;
+    uint64_t u;
+    struct {
+#if __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
+        uint64_t frac : 52;
+        uint64_t exp : 11;
+        uint64_t sign : 1;
+#else
+        uint64_t sign : 1;
+        uint64_t exp : 11;
+        uint64_t frac : 52;
+#endif
+    } parts;
+} doubleBits;
+
+void printSm2p(double x)
+{
+    doubleBits db;
+    db.d = x;
+    int sign = db.parts.sign;
+    uint64_t exp_bits = db.parts.exp;
```

Comment:
```suggestion
    uint64_t expBits = db.parts.exp;
```
Вы как-то стрёмно решили конфликты